lib/puppet/transaction/resource_harness.rb

```diff
@@ -120,7 +120,7 @@ def sync_if_needed(param, context)
     do_audit = context.audited_params.include?(param.name)
 
     begin
-      if !(param.should.nil? || param.safe_insync?(current_value))
+      if param.should && !param.safe_insync?(current_value)
```

Comment:
Can you squash this with the previous commit, so that the resource_harness is not modified in any commit?